### PR TITLE
Views for OnlineRecipe + Followees

### PR DESCRIPTION
**Pull request by chrystalquek**
- [x] Delete followee -> app crash [NOTE that some runs this happens, some runs theres no problem at all, problem cannot be replicated as of now]
- [x] Republish of recipe dosent work (somehow 
- [x] Modal dosent appear to download recipe
- [x] Creator name not currently displayed
- [x] once rate, all recipes refreshed @ same time and get pushed up to top [NOTE that some runs this happens, some runs theres no problem at all, problem cannot be replicated as of now]
- [x] NotFoundView not centralised to middle of page
- [x] Pass UserSettings to all ViewModels??
- [ ] OnlineRecipeView's totalTimeTaken (might be changed by graph implementation)
- [ ] Once delete local recipe, can nvr get back connection to online recipe [add download button to my published recipes?]. Check thru local recipes whether any contains onlineId
- [ ] Move OnlineRecipe stuff to a new folder
- [x] Need help in testing concurrency (eg 2 ppl rate recipe at same time)

### CODE REVIEW

File: ChopChop/ViewModels/Recipes/RecipeViewModel.swift [hunk 1]
```diff
@@ -97,4 +101,26 @@ class RecipeViewModel: ObservableObject {
...
+    private func bindPublished() {
+        recipe.$onlineId
+            .sink { [weak self] onlineId in
+                print(onlineId)
+                self?.isPublished = onlineId != nil
+            }
+            .store(in: &cancellables)
+    }
+
+    func publish() {
```
Comment (chrystalquek) [inline ChopChop/ViewModels/Recipes/RecipeViewModel.swift:108]: not working currently (can publish, cant republish)

File: ChopChop/ViewModels/UserProfile/CreateUserProfileViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import Combine
+import Foundation
+import UIKit
+
+var USER_ID = UserDefaults.standard.string(forKey: "userId")
```
Comment (chrystalquek) [inline ChopChop/ViewModels/UserProfile/CreateUserProfileViewModel.swift:5]: not sure where to put this

File: ChopChop/Storage/StorageManager.swift [hunk 3]
```diff
@@ -413,14 +428,33 @@ extension StorageManager {
         }
         var localRecipe = try Recipe(
             name: newName,
-            onlineId: recipe.id,
+            // if original creator, create a link to online recipe. otherwise nil
+            onlineId: recipe.userId == USER_ID ? recipe.userId : nil,
```
Comment (shadowezz) [inline ChopChop/Storage/StorageManager.swift]: Should it be recipe.id instead of recipe.userId

File: ChopChop/Models/User/CurrentUser.swift [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import Foundation
+
+var USER_ID = UserDefaults.standard.string(forKey: "userId")
```
Comment (shadowezz) [inline ChopChop/Models/User/CurrentUser.swift]: Is it better to store this in the UserSetting Object and use it as environmentObject that is accessible to all subviews. So like when the app starts, the user setting will get the id from UserDefaults and we just use the environment object from then on. Not sure if that is how environmentObject works.

**Review by shadowezz — COMMENTED** [review]
Some comments

**Pull request closed without merging**